post-form.js

```diff
@@ -0,0 +1,96 @@
+import {parseFilename} from './lib.js';
+
+export default class CreatePost extends HTMLElement {
+  constructor() {
+    super();
+    this._root = this.attachShadow({mode: 'open'})
+    this._root.innerHTML = `
+      <link href="index.css" rel="stylesheet">
+      <style>
+        div.formContainer {
+          display: flex;
+          flex-direction: column;
+          gap: 5px;
+          margin: 1em 0;
+        }
+      </style>
+      <form id="postForm"></form>
+      <div class="formContainer">
```

Comment:
Would it make sense to wrap all of this with the form instead of having the extra div?

Reply:
I did this since the file upload is in it's own form and forms can't be nested, but using the form attribute on inputs, you can achieve the desired functionality. eg
```html
<form id="form1"></form>
<form id="form2" enctype="multipart/form-data"></form>
<input name="field1form1" form="form1" />
<input name="field1form2" form="form2" />
```

I'm never 100% sure how to best use forms when doing client side apps. The most basic idea that makes sense to me is one form per POST. But since form logic are intercepted in most cases, it's entirely possible to have no forms at all.

Reply:
That makes sense, I hadn't considered the file upload.